### PR TITLE
Retry tavily outside of model parsing and allow to not generate subqueries

### PR DESCRIPTION
**Pull request by kongzii**

### CODE REVIEW

File: prediction_prophet/functions/research.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def research(
     logger: t.Union[logging.Logger, "Logger"] = logging.getLogger()
 ) -> str:
     logger.info("Started subqueries generation")
-    queries = generate_subqueries(query=goal, limit=initial_subqueries_limit, model=model, api_key=openai_api_key)
+    queries = generate_subqueries(query=goal, limit=initial_subqueries_limit, model=model, api_key=openai_api_key) if initial_subqueries_limit > 0 else [goal]
```
Comment (evangriffiths) [inline prediction_prophet/functions/research.py:32]: imo makes more sense to move this into the function definition - i.e. at the start:
```
if limit == 0:
  return [query]
```
so don't need to have this logic wherever it's used

**Review by evangriffiths — COMMENTED** [review]
Is this just in case there is a bug inside the `transformed_results = [...]` bit of the function?

File: prediction_prophet/functions/research.py (hunk 1 above)
Comment (kongzii) [reply prediction_prophet/functions/research.py]: Sure, no preference on my side

**Comment by kongzii** [comment]
> Is this just in case there is a bug inside the transformed_results = [...] bit of the function?

yeah, if they change output model, it won't help to retry the call

**Review by evangriffiths — APPROVED** [review]

**Pull request merged**